### ISSUE TITLE
Problems reconnecting!?

### ISSUE THREAD

**Issue by dCosminn** [pyrogram/pyrogram#103]
I was testing in VM on ubuntu 18 to see if the library would reconnect or give a error when the internet goes down but it won't reconnect or give errors. 

After a successful connection i kill the internet and get a bunch of this, which is normal
```
INFO:pyrogram.connection.connection:Connecting...
INFO:pyrogram.connection.connection:Connecting...
INFO:pyrogram.connection.connection:Disconnected
INFO:pyrogram.connection.connection:Connecting...
```

But then when i re-enable the internet it won't reconnect and i only get this:
```
INFO:pyrogram.connection.transport.tcp.tcp_abridged:Connected!
INFO:pyrogram.client.ext.syncer:Synced my_account
INFO:pyrogram.connection.connection:Disconnected
INFO:pyrogram.connection.connection:Connecting...
INFO:pyrogram.connection.transport.tcp.tcp_abridged:Connected!
INFO:pyrogram.client.ext.syncer:Synced my_account
INFO:pyrogram.connection.connection:Disconnected
INFO:pyrogram.connection.connection:Connecting...
INFO:pyrogram.connection.transport.tcp.tcp_abridged:Connected!
INFO:pyrogram.client.ext.syncer:Synced my_account

```
Is this normal, how would i be able to reconnect?

Sample code:
```
logging.basicConfig(level=logging.INFO)

app = Client(session_name, api_id, api_hash)

@app.on_message(Filters.private)
def hello(client, message):
    message.reply("Hello {}".format(message.from_user.first_name))
app.run()
```

**Comment by m3r3lin** [comment]
solution : for this issue change app.run() to app.start() :+1:

**Comment by delivrance** [comment]
> solution : for this issue change app.run() to app.start()

Well that's weird because what `app.run()` does is just call `app.start()` and `app.idle()`
https://github.com/pyrogram/pyrogram/blob/0c96fa8b7cd4a7f22bceacf12c85d5f68770db98/pyrogram/client/client.py#L324-L332

Can you really reproduce this consistently?

**Comment by dCosminn** [comment]
Yes, app.start() does not work for me.
Maybe he did something else to fix it?

**Comment by delivrance** [comment]
@dCosminn I just notices I was not replying to you. @m3r3lin's gravatar looks almost identical to yours.

So, does this happen inside a VM only? If so, how are you killing your connection exactly? What's you host OS and VM provider/application?

**Comment by dCosminn** [comment]
I kill the connection by disabling the network adapter on host OS or by disconnecting the network adapter on VM. 
Like with the connection ON I let the pyrogram to connect, then i disable the network adapter, wait 10-15 seconds and enable the network adapter.

Python 3.6.6
Pyrogram Version: 0.7.5
Host OS is Windows 10 Pro, Build 1809, it's a alpha build.. but it happens on stable builds too.
VMware player 12.5.6 build-5528349 and the guest OS is Windows 7 Home Premium, Build 7601

**Comment by delivrance** [comment]
Fixed in b619818c407c4ec298e563614360db7d811e0830.

You can upgrade Pyrogram with `pip3 install --upgrade git+https://github.com/pyrogram/pyrogram.git` to have it in your installation.

**Comment by Shams76** [comment]
> I was testing in VM on ubuntu 18 to see if the library would reconnect or give a error when the internet goes down but it won't reconnect or give errors.
> 
> After a successful connection i kill the internet and get a bunch of this, which is normal
> 
> ```
> INFO:pyrogram.connection.connection:Connecting...
> INFO:pyrogram.connection.connection:Connecting...
> INFO:pyrogram.connection.connection:Disconnected
> INFO:pyrogram.connection.connection:Connecting...
> ```
> 
> But then when i re-enable the internet it won't reconnect and i only get this:
> 
> ```
> INFO:pyrogram.connection.transport.tcp.tcp_abridged:Connected!
> INFO:pyrogram.client.ext.syncer:Synced my_account
> INFO:pyrogram.connection.connection:Disconnected
> INFO:pyrogram.connection.connection:Connecting...
> INFO:pyrogram.connection.transport.tcp.tcp_abridged:Connected!
> INFO:pyrogram.client.ext.syncer:Synced my_account
> INFO:pyrogram.connection.connection:Disconnected
> INFO:pyrogram.connection.connection:Connecting...
> INFO:pyrogram.connection.transport.tcp.tcp_abridged:Connected!
> INFO:pyrogram.client.ext.syncer:Synced my_account
> ```
> 
> Is this normal, how would i be able to reconnect?
> 
> Sample code:
> 
> ```
> logging.basicConfig(level=logging.INFO)
> 
> app = Client(session_name, api_id, api_hash)
> 
> @app.on_message(Filters.private)
> def hello(client, message):
>     message.reply("Hello {}".format(message.from_user.first_name))
> app.run()
> ```

Your country block with telegram  use vpn for connect
